mcs/class/System/System.Diagnostics/Process.cs

```diff
@@ -606,8 +606,7 @@ bool StartWithShellExecuteEx (ProcessStartInfo startInfo)
 				throw new Win32Exception (-proc_info.pid);
 			}
 
-			m_processHandle = new SafeProcessHandle (proc_info.process_handle, true);
-			haveProcessHandle = true;
+			SetProcessHandle (new SafeProcessHandle (proc_info.process_handle, true));
 			SetProcessId (proc_info.pid);
 
 			if (watchForExit)
```

Comment:
This won't be necessary when you switch to the other SetProcessHandle since it does it internally already.